src/http/internal_state.zig

```diff
@@ -0,0 +1,248 @@
+pub const HTTPStage = enum {
```

Comment:
The file name for this should be `InternalState.zig` and InternalState should be a top-level struct.